modyn/models/articlenet/articlenet.py

```diff
@@ -0,0 +1,50 @@
+from typing import Any
+
+import torch
+from torch import nn
+from transformers import DistilBertForSequenceClassification, DistilBertModel
+
+
+class ArticleNet:
+    # pylint: disable-next=unused-argument
+    def __init__(self, model_configuration: dict[str, Any], device: str, amp: bool) -> None:
+        self.model = ArticleNetwork(**model_configuration)
+        self.model.to(device)
+
+
+class DistilBertClassifier(DistilBertForSequenceClassification):
+    def __call__(self, data: torch.Tensor) -> torch.Tensor:
+        input_ids = data[:, :, 0]
+        attention_mask = data[:, :, 1]
+        outputs = super().__call__(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+        )[0]
+        return outputs
+
+
+class DistilBertFeaturizer(DistilBertModel):
+    def __init__(self, config: Any) -> None:
+        super().__init__(config)
+        self.d_out = config.hidden_size
+
+    def __call__(self, data: torch.Tensor) -> torch.Tensor:
+        input_ids = data[:, :, 0]
+        attention_mask = data[:, :, 1]
+        hidden_state = super().__call__(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+        )[0]
+        pooled_output = hidden_state[:, 0]
+        return pooled_output
+
+
+class ArticleNetwork(nn.Module):
```

Comment:
Could you add some documentation to the models, i.e., where are they coming from? Is the term ´ArticleNetwork´ something I/a user should know? Where is that coming from, why are we using a distilbeter featurizer before some linear layer? Usually, when building a very simple NLP DNN, huggingface is not necesasry, right? Why do we need/use it here?

Reply:
The models and names are taken from Wild-time

Reply:
I have added some documentation. To answer your questions:
- Wildtime
- No, it is the name used for the two NLP tasks that both deal with articles
- Typically classification for NLP tasks uses a featurizer (in this case DistilBert) to exploit the [CLS] (classification) token and from there project with a linear layer. See [here](https://towardsdatascience.com/text-classification-with-bert-in-pytorch-887965e5820f) for example
- I don't think it is a _very simple NLP DNN_, it has about 60 million trainable parameters
- Same reason we use torchvision for resnet

Reply:
I didn't realize at first we were actually using DistilBert, makes sense. Thank you.